src/utils.jl

```diff
@@ -400,3 +399,69 @@ function record_erupt_end(
     push!(erupt_saved.volume, volume)
     return nothing
 end
+
+"""
+    ic_phase_conversion(phase_here::T, composition::Union{Silicic,Mafic}, M_h2o::T, M_co2::T, M_tot::T, P::T, Temp::T, V::T, rho_m::T, param_IC::ParamICFinder{T})::NamedTuple{(:eps_g_temp, :X_co2_temp, :C_co2_temp, :phase),NTuple{4,T}} where {T<:Float64}
+
+This function is used to handle phase conversion by iteratively modifying the `max_count` or `Tol` parameters of function `IC_Finder` until the correct phase is obtained.
+- This function is used within the `affect!` function.
+"""
+function ic_phase_conversion(
+    phase_here::T,
+    composition::Union{Silicic,Mafic},
+    M_h2o::T,
+    M_co2::T,
+    M_tot::T,
+    P::T,
+    Temp::T,
+    V::T,
+    rho_m::T,
+    param_IC::ParamICFinder{T},
+)::NamedTuple{
+    (:eps_g_temp, :X_co2_temp, :C_co2_temp, :phase),NTuple{4,T}
+} where {T<:Float64}
+    eps_g_temp, X_co2_temp, C_co2_temp, phase = IC_Finder(
+        composition, M_h2o, M_co2, M_tot, P, Temp, V, rho_m, param_IC
+    )
+    @info(
+        "1st attempt in IC Finder $(phase_here != phase ? "successful\n  phase_here: $phase_here, new_phase: $phase" : "unsuccessful")"
+    )
+    if phase_here != phase
+        return (; eps_g_temp, X_co2_temp, C_co2_temp, phase)
+    else
+        param_IC.max_count = 150
+        eps_g_temp, X_co2_temp, C_co2_temp, phase = IC_Finder(
+            composition, M_h2o, M_co2, M_tot, P, Temp, V, rho_m, param_IC
+        )
+        param_IC.max_count = 100
+        @info(
+            "2nd attempt in IC Finder $(phase_here != phase ? "successful\n  phase_here: $phase_here, new_phase: $phase" : "unsuccessful")"
+        )
+        if phase_here != phase
+            return (; eps_g_temp, X_co2_temp, C_co2_temp, phase)
+        else
+            i = 3
+            i_max = 6   # max iteration of IC_Finder
+            while phase_here == phase && i <= i_max
+                param_IC.Tol = param_IC.Tol * 0.1
+                eps_g_temp, X_co2_temp, C_co2_temp, phase = IC_Finder(
+                    composition, M_h2o, M_co2, M_tot, P, Temp, V, rho_m, param_IC
+                )
+                @info(
+                    "$i$(i == 3 ? "rd" : "th") attempt in IC Finder $(phase_here != phase ? "successful\n  phase_here: $phase_here, new_phase: $phase" : "unsuccessful")"
+                )
+                if phase_here != phase
+                    param_IC.Tol = param_IC.Tol * 10^(i - 2)
+                    break
+                elseif i == i_max
+                    param_IC.Tol = param_IC.Tol * 10^(i - 2)
+                    @warn(
+                        "phase conversion unsuccessful, IC_Finder results:\n  $eps_g_temp, $X_co2_temp, $C_co2_temp, $phase = IC_Finder($composition, $M_h2o, $M_co2, $M_tot, $P, $Temp, $V, $rho_m, $param_IC"
+                    )
+                end
+                i += 1
+            end
```

Comment:
Here is the addition part comparing the previous version of dealing with phase conversion. I found it need more iterative changes of parameters `Tol` to let phase successfully conversion after fixing corresponding changes from Matlab code.

Reply:
For example, here's a eps_g result after fixing corresponding changes from Matlab code while it shouldn't be negative. That's because phase conversion was unsuccessful around the end.
![image](https://github.com/brown-ccv/Chamber.jl/assets/62089495/7b113757-ed47-4a54-b05f-e4cc920f1269)

By adding more iterative changes of `Tol` to let phase change:
![image](https://github.com/brown-ccv/Chamber.jl/assets/62089495/dbf5a9e7-ce8c-40bc-b736-df80f1059d12)

Reply:
@CallieHsu I mentioned this [elsewhere](https://github.com/brown-ccv/Chamber.jl/issues/246#issuecomment-1634625746). What do you think of creating a repo for the MATLAB code?

Reply:
@cpaniaguam I think it is fine. Do the repo for MATLAB code need to be public?

Reply:
@CallieHsu I don't see why it shouldn't be public. And I still think the group needs its own GitHub organization to host all the software generated by the group. I will see about creating one. Could it be named _HuberLab_, @chris-huber?